File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(const std::vector<int>& nums) {
        std::bitset<101> seen = {};
        int sum = std::numeric_limits<int>::min();
        for (int value : nums) {
            if (value > 0 && !seen[value]) {
                seen.set(value);
                if (sum < 0) {
                    sum = value;
                } else {
                    sum += value;
                }
            } else if (sum < 0 && value > sum) {
                sum = value;
            }
        }
        return sum;
    }
};

File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.py
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        if max(nums)<0:
            return max(nums)
        elif len(nums)==1:
            return nums[0]
        else:
            s=set()
            for i in range(len(nums)):
                if nums[i]>0:
                    s.add(nums[i])
            return sum(s)